internal/policy/targets.go

```diff
@@ -96,6 +102,54 @@ func UpdateDelegation(targetsMetadata *tuf.TargetsMetadata, ruleName string, aut
 	return targetsMetadata, nil
 }
 
+// ReorderDelegations changes the order of delegations in TargetsMetadata.
+func ReorderDelegations(targetsMetadata *tuf.TargetsMetadata, ruleNames []string) (*tuf.TargetsMetadata, error) {
```

Comment:
I think this could generally be simplified a bit... cc @patzielinski

Reply:
I haven't figure out a way to do this, but all other requested changes have been made.

Reply:
Taking a stab at making this easier to read:


```
func ReorderDelegations(targetsMetadata *tuf.TargetsMetadata, ruleNames []string) (*tuf.TargetsMetadata, error) {
	// Create a map of all existing delegations for quick look up
	rolesMap := make(map[string]tuf.Delegation)
	// Create a set of current rules in metadata, skipping the allow rule
	currentRules := set.NewSet[string]()
	for _, delegation := range targetsMetadata.Delegations.Roles {
		if delegation.Name == AllowRuleName {
			continue
		}
		rolesMap[delegation.Name] = delegation
		currentRules.Add(delegation.Name)
	}

	specifiedRules := set.NewSet[string]()
	for _, name := range ruleNames {
		if specifiedRules.Has(name) {
			return nil, fmt.Errorf("%w: '%s'", ErrDuplicatedRuleName, name)
		}
		specifiedRules.Add(name)
	}

	onlyInSpecifiedRules := specifiedRules.Minus(currentRules)
	if onlyInSpecifiedRules.Len() != 0 {
		if onlyInSpecifiedRules.Has(AllowRuleName) {
			return nil, fmt.Errorf("%w: do not specify allow rule", ErrCannotManipulateAllowRule)
		}

		contents := onlyInSpecifiedRules.Contents()
		return nil, fmt.Errorf("%w: rules '%s' do not exist in current rule file", ErrRuleNotFound, strings.Join(contents, ", "))
	}

	onlyInCurrentRules := currentRules.Minus(specifiedRules)
	if onlyInCurrentRules.Len() != 0 {
		contents := onlyInCurrentRules.Contents()
		return nil, fmt.Errorf("%w: rules '%s' not specified", ErrRuleNotFound, strings.Join(contents, ", "))
	}

	// Create newDelegations and set it in the targetsMetadata after adding allow rule
}
```

I think this may be slightly less optimal as we will end up iterating over `ruleNames` at least twice directly and some more in the implementation of `set.Minus` (which I haven't shown here), but I think that's okay because:
a) reordering is unlikely to be a regular operation
b) some of the cost (eg. `set.Contents()`) is going to only be during error cases

I personally think the result ends up being more readable.

Reply:
We could also only do the `.Minus` when `specifiedRules` and `currentRules` are not equal, as `.Minus` is really only needed for the error message to be helpful.

Reply:
I haven't found the implementation for `.Minus`, so I added a quick impl in the common/set package